core/file_test.go

```diff
@@ -69,6 +70,62 @@ func TestLoadTomlFile_FileExitsShouldPass(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestSaveTomlFile(t *testing.T) {
+	t.Parallel()
+
+	t.Run("empty filename, should fail", func(t *testing.T) {
+		t.Parallel()
+
+		cfg := &TestStruct{}
+
+		fileName := ""
+
+		err := core.SaveTomlFile(cfg, fileName)
+		require.Error(t, err)
+	})
+
+	t.Run("invalid path, should fail", func(t *testing.T) {
+		t.Parallel()
+
+		cfg := &TestStruct{}
+
+		fileName := "invalid_path" + "/testFile1"
+
+		err := core.SaveTomlFile(cfg, fileName)
+		require.Error(t, err)
+	})
+
+	t.Run("should work with valid dir", func(t *testing.T) {
+		t.Parallel()
+
+		cfg := &TestStruct{}
+
+		dir := t.TempDir()
+		fileName := dir + "/testFile1"
+
+		err := core.SaveTomlFile(cfg, fileName)
+		if _, errF := os.Stat(fileName); errF == nil {
+			_ = os.Remove(fileName)
+		}
+
+		require.Nil(t, err)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		cfg := &TestStruct{}
+		fileName := "testFile2"
+
+		err := core.SaveTomlFile(cfg, fileName)
+		if _, errF := os.Stat(fileName); errF == nil {
+			_ = os.Remove(fileName)
+		}
+
+		assert.Nil(t, err)
+	})
+}
+
```

Comment:
maybe add an "integration test" between the SaveTomlFile and LoadTomlFile? Create a mock struct, add some fields, save it, load it, and compare expected vs loaded structs. Should not find errors in the process.

Reply:
updated test to include save and load